src/uproot/source/futures.py

```diff
@@ -469,3 +472,45 @@ def __exit__(self, exception_type, exception_value, traceback):
         self.shutdown()
         self._resource.__exit__(exception_type, exception_value, traceback)
         self._closed = True
+
+
+class LoopExecutor:
+    def __repr__(self):
+        return f"<LoopExecutor at 0x{id(self):012x}>"
+
+    def __init__(self):
+        self._loop = asyncio.new_event_loop()
+        # TODO: remove daemon=True (or not?)
+        self._thread = threading.Thread(target=self._run, daemon=True)
```

Comment:
I don't think we want a `daemon` here — we want fsspec to be able to gracefully close  its resources if possible.

Reply:
Yeah I agree, it should be possible to do this cleanup correctly